modules/http/src/clients/axios-http-client.ts

```diff
@@ -0,0 +1,199 @@
+ * httpClient.init(true, onRequestStart, onRequestSuccess, onRequestError, onRequestFinish);
+ * ```
+ */
+@staticDecorator<HttpClientStatic<AxiosInstance>>()
+export class AxiosHttpClient implements HttpClient<AxiosRequestConfig, AxiosResponse, AxiosError> {
+
+    private static axiosInstance: AxiosInstance;
+    private static clientInstance: AxiosHttpClient;
+    private isHandlerEnabled: boolean;
+    private requestStartCallback: () => void;
+    private requestSuccessCallback: (response: AxiosResponse) => void;
+    private requestErrorCallback: (error: AxiosError) => void;
+    private requestFinishCallback: () => void;
+
+    /**
+     * Private constructor to avoid object instantiation from outside
+     * the class.
+     *
+     * @hideconstructor
+     */
+    private constructor() {
+        this.init = this.init.bind(this);
+    }
+
+    /**
+     * Returns an aggregated instance of type `AxiosInstance` of `AxiosHttpClient`.
+     *
+     * @return {any}
+     */
+    public static getInstance(): AxiosInstance & any {
+        if (!this.axiosInstance) {
+            this.axiosInstance = axios.create();
+        }
+
+        if (!this.clientInstance) {
+            this.clientInstance = new AxiosHttpClient();
+        }
+
+        // Register request interceptor
+        this.axiosInstance.interceptors.request.use(
+            (request) => this.clientInstance.requestHandler(request)
+        );
+
+        // Register response interceptor
+        this.axiosInstance.interceptors.response.use(
+            (response) => this.clientInstance.successHandler(response),
+            (error) => this.clientInstance.errorHandler(error)
+        );
+
+        return { ...this.axiosInstance, ...this.clientInstance };
+    }
+
+    /**
+     * Intercepts all the requests.
+     * If the `isHandlerEnabled` flag is set to true, fires the `requestStartCallback`
+     * and retrieves the access token from the server and attaches it to the request.
+     * Else, just returns the original request.
+     *
+     * @param {AxiosRequestConfig} request - Original request.
+     * @return {AxiosRequestConfig}
+     */
+    public requestHandler(request: AxiosRequestConfig): AxiosRequestConfig {
+        if (this.isHandlerEnabled) {
+            if (this.requestStartCallback && typeof this.requestStartCallback === "function") {
+                this.requestStartCallback();
+            }
+            return AuthenticateSessionUtil.getAccessToken()
+                .then((token) => {
+                    request.headers.Authorization = `Bearer ${ token }`;
+                    return request;
+                })
+                .catch((error) => {
+                    if (this.requestFinishCallback && typeof this.requestFinishCallback === "function") {
+                        this.requestFinishCallback();
+                    }
+                    return Promise.reject(`Failed to retrieve the access token: ${error}`);
```

Comment:
Missing spaces between curly braces: `${error}`

Reply:
Added the spaces between curly brackets.